src/moclojer/core.clj

```diff
@@ -17,22 +18,58 @@
 
 (defn handler
   "prepare function to receive http request (handler)"
-  [r]
-  [(body-params/body-params)
-   http/json-body
-   (fn [req]
-     {:status       (get-in r [:endpoint :response :status] 200)
-      :content-type (get-in r [:endpoint :response :headers :content-type]
-                            "application/json")
-      :body         (selmer/render (get-in r [:endpoint :response :body] "{}")
-                                   {:path-params  (:path-params req)
-                                    :query-params (:query-params req)
-                                    :json-params  (:json-params req)})})])
+  ([r]
+   [(body-params/body-params)
+    http/json-body
+    (fn [req]
+      {:status       (get-in r [:endpoint :response :status] 200)
+       :content-type (get-in r [:endpoint :response :headers :content-type]
+                             "application/json")
+       :body         (selmer/render (get-in r [:endpoint :response :body] "{}")
+                                    {:path-params  (:path-params req)
+                                     :query-params (:query-params req)
+                                     :json-params  (:json-params req)})})])
+  ([r route-name]
+   [(body-params/body-params)
+    http/json-body
+    (fn [req]
+      {:status       (get-in r [route-name :response :status] 200)
+       :content-type (get-in r [route-name :response :headers :content-type]
+                             "application/json")
+       :body         (selmer/render (get-in r [route-name :response :body] "{}")
+                                    {:path-params  (:path-params req)
+                                     :query-params (:query-params req)
+                                     :json-params  (:json-params req)})})]))
+
+(defmulti make-router (fn [config]
+                        (when (= (get config "openapi") "3.0.0") :open-api)
```

Comment:
- This function never returns `:open-api`
- Not a good pattern have to extend the dispatch-fn to extend the multimethod.

Reply:
I agreed.
do you have any recommendations, how could I continue with it?

Reply:
@souenzzo  what do you think in create 3 functions
make-edn-route
make-yml-route
make-open-api-route ?
instead of use a defmult/defmethod?